lib/smart_proxy_container_gateway/container_gateway_main.rb

```diff
@@ -40,6 +40,16 @@ def pulp_registry_request(uri, headers)
         end
       end
 
+      def flatpak_static_index(headers, params = {})
+        query = ""
+        unless params.empty?
+          query_parts = params.map { |key, value| "#{key}=#{value}" }
+          query = "?#{query_parts.join('&')}"
+        end
+        uri = URI.parse("#{@pulp_endpoint}/pulpcore_registry/index/static#{query}")
```

Comment:
I'd like the parameters to be escaped. I'd suggest this:
```suggestion
        uri = URI.parse("#{@pulp_endpoint}/pulpcore_registry/index/static")
        uri.query = params.map { |k, v| "#{URI.encode_uri_component(k)}=#{URI.encode_uri_component(v)}" }.join('&') if params.any?
```

I really wish there was a method similar to `URI.encode_www_form` that uses `URI.encode_uri_component` and accepts a hash, but I can't find it.

And the `tags` method probably needs the same fixes.

Reply:
I swear I saw it used somewhere because I too thought there must be an easy way, but I couldn't find it. Feels so stupid because it's so extremely common.

Reply:
I wonder if the + that `encode_www_form` is okay.